asn1ate/pyasn1gen.py

```diff
@@ -567,23 +574,37 @@ def _sanitize_identifier(name):
     return name
 
 
+def _sanitize_module(name):
+    """ Sanitize ASN.1 module identifiers so that they're PEP8 compliant identifiers.
+    """
+    return _sanitize_identifier(name).lower()
+
 # Simplistic command-line driver
-def main(args):
-    with open(args[0]) as f:
-        asn1def = f.read()
+def main():
+    arg_parser = argparse.ArgumentParser(description='Generate Python classes from an ASN1 definition file. Output to stdout by default.')
+    arg_parser.add_argument('filename', metavar='filename', type=argparse.FileType('r'),
+                        help='the ASN1 file to process')
+    arg_parser.add_argument('--split', action='store_true',
+                        help='output multiple modules to separate files')
+    args = arg_parser.parse_args()
+    asn1def = args.filename.read()
 
     parse_tree = parser.parse_asn1(asn1def)
 
     modules = build_semantic_model(parse_tree)
-    if len(modules) > 1:
+    if len(modules) > 1 and not args.split:
         print('WARNING: More than one module generated to the same stream.', file=sys.stderr)
 
+    output_file = sys.stdout
     for module in modules:
-        print(pygen.auto_generated_header())
-        generate_pyasn1(module, sys.stdout)
+        module.referenced_modules = [m for m in modules if not m is module]
+        if args.split:
+            output_file = open(_sanitize_module(module.name) + '.py', 'w')
```

Comment:
I'd like it if this file handle was closed properly. Conditional file vs. stdout is tricky in Python with context managers, so I think it's hard to do this nicely... How about:

    try:
        # main logic
    finally:
        if output_file != sys,stdout:
            output_file.close()

?